pyiceberg/table/__init__.py

```diff
@@ -160,7 +160,9 @@ def _check_schema_compatible(table_schema: Schema, other_schema: "pa.Schema") ->
     """
     Check if the `table_schema` is compatible with `other_schema`.
 
-    Two schemas are considered compatible when they are equal in terms of the Iceberg Schema type.
+    The schemas are compatible if:
```

Comment:
This is a nit, but should we keep the bit about the comparison being between the `table_schema` and the Iceberg schema representation of `other_schema`?